src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,56 @@
 package core.basesyntax;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> dataFromFile = readFromFile(fromFileName);
+        int supplyAmount = calculateTotal(dataFromFile, SUPPLY);
+        int buyAmount = calculateTotal(dataFromFile, BUY);
+        int result = supplyAmount - buyAmount;
+        List<String> report = createReport(supplyAmount, buyAmount, result);
+        writeToFile(toFileName, report);
+    }
+
+    private List<String> readFromFile(String fileName) {
+        try {
```

Comment:
It's better to use try-with-resources here with BufferedReader for closing resources automatically